### PR TITLE
Update codebase to ZAP 2.11

### PR DESCRIPTION
**Pull request by thc202**
Change all add-ons and `testutils` to use 2.11 (SNAPSHOT).
Update code accordingly (e.g. add annotations, address deprecations).

### CODE REVIEW

**Review by psiinon — APPROVED** [review]

File: addOns/addOns.gradle.kts [hunk 1]
```diff
@@ -117,7 +117,12 @@ subprojects {
         }
     }
 
-    val apiGenClasspath = configurations.detachedConfiguration(dependencies.create("org.zaproxy:zap:2.10.0"))
+    val zapGav = "org.zaproxy:zap:2.11.0-20210929.165234-4"
+    dependencies {
+        "zap"(zapGav)
```
Comment (kingthorin) [inline addOns/addOns.gradle.kts:122]: Gav? Generally available version?

**Review by kingthorin — APPROVED** [review]

File: addOns/addOns.gradle.kts (hunk 1 above)
Comment (thc202) [reply addOns/addOns.gradle.kts:122]: "group, artifact, version"

**Comment by thc202** [comment]
Addressed TODOs related to 2.11.

File: addOns/gettingStarted/CHANGELOG.md [hunk 2]
```diff
@@ -11,12 +11,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## [12] - 2020-12-15
 ### Changed
 - Update link to OWASP ZAP homepage.
-- Updated for 2.10.0
 - Update minimum ZAP version to 2.10.0.
```
Comment (kingthorin) [inline addOns/gettingStarted/CHANGELOG.md:14]: 2.11?

**Review by kingthorin — APPROVED** [review]
Looks okay to me.

**Comment by thc202** [comment]
Going to rebase to pick the other alert tag changes (which will require the override).

**Comment by kingthorin** [comment]
Thanks!

**Comment by thc202** [comment]
Done.

**Review by kingthorin — APPROVED** [review]
LGTM!

**Review by psiinon — APPROVED** [review]
Me too :)

**Comment by kingthorin** [comment]
FYI, builds might be stuck: https://www.githubstatus.com/incidents/bdbzpz7qxmbx

**Pull request merged**